net/pdnsd/files/pdnsd.init

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2019 Xiaobo Tian
+# Copyright (C) 2001-2001 Thomas Moestl
+# Copyright (C) 2002-2010 Paul A. Rombouts
+
+USE_PROCD=1
+
+START=65
+STOP=10
+
+PROG=/usr/sbin/pdnsd
+CONFFILE=/etc/pdnsd.conf
+CACHEDIR=/tmp/pdnsd
+CACHEFILE=$CACHEDIR/pdnsd.cache
+
+start_service() {
+	if [ ! -d "$CACHEDIR" ]; then
+		mkdir -p $CACHEDIR
+		echo -ne "pd13\000\000\000\000" > $CACHEFILE
```

Comment:
Now I am being curious again.  Does pdnsd really cannot handle nonexistence of the cache file?  Do we really need to seed it with "pd13" plus 4 more null bytes?

Reply:
/* A version identifier to prevent reading incompatible cache files */
static const char cachverid[] = {'p','d','1','3'};

Reply:
using echo with flags (-ne) will trigger a shellcheck warning.

Can be avoided by using printf, eg:

```
printf "%s\000\000\000\000" "pd13" > $CACHEFILE
```

Reply:
Fixed

Reply:
if `$CACHEDIR` does not exist at all, mkdir should be enough right?  No need to initialize the file content?

Reply:
I think remove the initialization of the cache without any bad effects.